### PR TITLE
[Added] Loading of individual country instead of data from all countries

### PR DESCRIPTION
**Pull request by helloitsm3**
Tried to implement loading of single country instead of getting data from all countries and cities. Helps to reduce the time taken when initializing the app

### CODE REVIEW

**Review by tomayac — CHANGES_REQUESTED** [review]
Thanks for the proposal. At the moment it looks like this is hardcoding the country to `SG`, though. If this becomes a feature, it should be configurable at initialization time.

**Comment by helloitsm3** [comment]
> Thanks for the proposal. At the moment it looks like this is hardcoding the country to `SG`, though. If this becomes a feature, it should be configurable at initialization time.

I was using "SG" to test out to see if it was able to grab only SG.txt. Users will need to specify the country they plan to grab and it'll dump the data into geonames_dump folder

**Comment by tomayac** [comment]
Yes, but your PR right now hardcodes this.

**Comment by helloitsm3** [comment]
> Yes, but your PR right now hardcodes this.

Sorry, my bad. I'll remove it and make another pull request

**Comment by tomayac** [comment]
The way this works should be: 

```js
const geocoder = require('local-reverse-geocoder');
geocoder.init(
  {
    load: {
      admin1: true,
      admin2: false,
      admin3And4: false,
      alternateNames: false,
    },
    countries: ['SG'], // <== New, defaults to all countries
  },
  function () {
    // Ready to call lookUp
  }
);
```

**Comment by helloitsm3** [comment]
> The way this works should be: 
> 
> 
> 
> ```js
> 
> const geocoder = require('local-reverse-geocoder');
> 
> geocoder.init(
> 
>   {
> 
>     load: {
> 
>       admin1: true,
> 
>       admin2: false,
> 
>       admin3And4: false,
> 
>       alternateNames: false,
> 
>     },
> 
>     countries: ['SG'], // <== New, defaults to all countries
> 
>   },
> 
>   function () {
> 
>     // Ready to call lookUp
> 
>   }
> 
> );
> 
> ```

Okay thanks. I'll make the necessary changes later

File: app.js [hunk 1]
```diff
@@ -51,8 +51,19 @@ var port = Number(process.env.PORT || 3000);
...
-  });
+  geocoder.init(
+    {
+      load: {
+        admin1: true,
+        admin2: false,
+        admin3And4: false,
+        alternateNames: false,
+      },
+      countries: ['SG', 'AU', 'US'], // <== New, defaults to all countries if array is empty
```
Comment (tomayac) [inline app.js:62]:
```suggestion
      countries: [],
```

Comment (tomayac) [reply app.js:62]: Else, everyone who downloads the library will request your country defaults.

**Review by tomayac — COMMENTED** [review]
Could you add documentation in the README how people would use this, please?

**Comment by helloitsm3** [comment]
> Could you add documentation in the README how people would use this, please?

Yes. I'll work on that and push it later

**Review by tomayac — APPROVED** [review]
LGTM, approval.

Thanks for this. I'll make some minor adjustments and will release a new build soon.

**Comment by tomayac** [comment]
Fixes https://github.com/tomayac/local-reverse-geocoder/issues/23. 

The release just went out as v0.11.0. Thanks again! You're also credited in the README :-)

**Pull request merged**